src/es/ikigaimangas/src/eu/kanade/tachiyomi/extension/es/ikigaimangas/IkigaiMangas.kt

```diff
@@ -0,0 +1,219 @@
+package eu.kanade.tachiyomi.extension.es.ikigaimangas
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.interceptor.rateLimitHost
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import uy.kohesive.injekt.injectLazy
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.TimeZone
+
+class IkigaiMangas : HttpSource() {
+
+    override val baseUrl: String = "https://ikigaimangas.com"
+    private val apiBaseUrl: String = "https://panel.ikigaimangas.com"
+    private val pageViewerUrl: String = "https://ikigaitoon.com"
+
+    override val lang: String = "es"
+    override val name: String = "Ikigai Mangas"
+
+    override val supportsLatest: Boolean = true
+
+    override val client = super.client.newBuilder()
+        .rateLimitHost(baseUrl.toHttpUrl(), 1, 2)
+        .rateLimitHost(apiBaseUrl.toHttpUrl(), 2, 1)
+        .build()
+
+    override fun headersBuilder() = super.headersBuilder()
+        .add("Referer", "$baseUrl/")
+
+    private val json: Json by injectLazy()
+
+    private val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z'", Locale.US).apply {
+        timeZone = TimeZone.getTimeZone("UTC")
+    }
+
+    override fun popularMangaRequest(page: Int): Request {
+        val apiUrl = "$apiBaseUrl/api/swf/series?page=$page&column=view_count&direction=desc".toHttpUrl().newBuilder()
+            .build()
```

Comment:
No need to make it HttpUrl
```suggestion
        val apiUrl = "$apiBaseUrl/api/swf/series?page=$page&column=view_count&direction=desc"
```